packages/lib/src/prod/shared-production.ts

```diff
@@ -34,6 +43,86 @@ export function prodSharedPlugin(
   let isRemote
   const id2Prop = new Map<string, any>()
 
+  let federation_fn_import_id = ''
+
+  const transformImportFn = function (
+    this: PluginContext,
+    code,
+    chunk: OutputChunk
+  ) {
+    const ast = this.parse(code)
+    const magicString = new MagicString(code)
+    let importSharedRequired = false
+    let importSharedFound = false
+    let modified = false
+
+    walk(ast, {
+      enter(node: any) {
+        if (node.type === 'ImportDeclaration') {
+          if (
+            node.source.value.includes('__federation_fn_import-') &&
+            node.specifiers.some(
+              (specify) => specify.imported?.name === 'importShared'
+            )
+          ) {
+            importSharedFound = true
+          }
+
+          const sharedName = getSharedName(node.source.value)
+          if (sharedName) {
+            const declaration: (string | never)[] = []
+            if (!node.specifiers?.length) {
+              //  invalid import , like import './__federation_shared_lib.js' , and remove it
+              magicString.remove(node.start, node.end)
+              modified = true
+            } else {
+              node.specifiers.forEach((specify) => {
+                declaration.push(
+                  `${
+                    specify.imported?.name
+                      ? `${
+                          specify.imported.name === specify.local.name
+                            ? specify.local.name
+                            : `${specify.imported.name}:${specify.local.name}`
+                        }`
+                      : `default:${specify.local.name}`
+                  }`
+                )
+              })
+            }
+            if (declaration.length) {
+              magicString.overwrite(
+                node.start,
+                node.end,
+                `const {${declaration.join(
+                  ','
+                )}} = await importShared('${sharedName}');\n`
+              )
+              importSharedRequired = true
+              modified = true
+            }
+          }
+        }
+      }
+    })
+    if (importSharedRequired && !importSharedFound) {
+      magicString.prepend(
+        `import {importShared} from '/${this.getFileName(
```

Comment:
In remotes, if the shared lib is not generated, this import will not work.
I had to replace it with
`'./${this.getFileName(`
to fix it.
But this PR does resolve a issues I had where react was loaded multiple times, great job.

Reply:
I'm not sure if I can follow.
`getFileName` includes the asset directory so for me the import ends up being.

`<HOST>/asset/asset/__federation_fn_import-<HASH>.js` 

I did change the code to replace the asset directory with a dot to get a relative import. But I don't know it that addresses the problem that you are having. Help is appreciated :)